src/Spark.Engine/Service/FhirService.cs

```diff
@@ -339,42 +344,47 @@ public FhirResponse History(IKey key, HistoryParameters parameters)
             }
             IHistoryService historyExtension = this.GetFeature<IHistoryService>();
 
-            return CreateSnapshotResponse(historyExtension.History(key, parameters));
+            var h = await historyExtension.History(key, parameters);
+            return await CreateSnapshotResponse(h);
         }
 
-        public FhirResponse Mailbox(Bundle bundle, Binary body)
+        public Task<FhirResponse> Mailbox(Bundle bundle, Binary body)
         {
             throw new NotImplementedException();
         }
 
-        public FhirResponse CapabilityStatement(string sparkVersion)
+        public async Task<FhirResponse> CapabilityStatement(string sparkVersion)
         {
+            await System.Threading.Tasks.Task.CompletedTask;
```

Comment:
This one does need to be here, or something similar does (see below). The interface has this function async for potential implementers to use but this implementation doesn't use/need it. In order to prevent a compiler warning this type of call is used. The other option would be to not make the function awaitable and instead return using `Task.FromResult()`, which wraps the input in a completed Task.

```csharp
    public Task<FhirResponse> CapabilityStatement(string sparkVersion)
    {
        ICapabilityStatementService capabilityStatementService = this.GetFeature<ICapabilityStatementService>();
        var r = Respond.WithResource(capabilityStatementService.GetSparkCapabilityStatement(sparkVersion));
        return Task.FromResult(r);
    }	        
```